From 1abf9cb2d81319147014229f8034031aec2ca3c3 Mon Sep 17 00:00:00 2001
From: Jianhua Lu <lujianhua000@gmail.com>
Date: Tue, 24 Dec 2024 23:32:21 +0800
Subject: [PATCH] usb: typec: tcpm: add hack for xiaomi charger

---
 drivers/usb/typec/tcpm/tcpm.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/usb/typec/tcpm/tcpm.c b/drivers/usb/typec/tcpm/tcpm.c
index 6021eeb903fec..dd813e1da20d5 100644
--- a/drivers/usb/typec/tcpm/tcpm.c
+++ b/drivers/usb/typec/tcpm/tcpm.c
@@ -5,6 +5,7 @@
  * USB Power Delivery protocol stack.
  */
 
+#include "linux/printk.h"
 #include <linux/completion.h>
 #include <linux/debugfs.h>
 #include <linux/device.h>
@@ -1908,6 +1909,13 @@ static int tcpm_pd_svdm(struct tcpm_port *port, struct typec_altmode *adev,
 	tcpm_log(port, "Rx VDM cmd 0x%x type %d cmd %d len %d",
 		 p[0], cmd_type, cmd, cnt);
 
+	/* hack cmd to DISCOVER_SVID */
+	if (port->partner_ident.id_header == 0) {
+		cmd = CMD_DISCOVER_SVID;
+	}
+	pr_err("Rx VDM cmd 0x%x type %d cmd %d len %d", p[0], cmd_type, cmd, cnt);
+	pr_err("charger id: 0x%x", port->partner_ident.id_header);
+
 	switch (rx_sop_type) {
 	case TCPC_TX_SOP_PRIME:
 		modep_prime = &port->mode_data_prime;
-- 
GitLab

